### PR TITLE
Add Sync Project Endpoint

### PR DESCRIPTION
**Pull request by Dostonbek1**
## Description 

Adds an API endpoint to sync an existing project.

- refactor importing project files code
- add sync project endpoint `api/projects/<project_id>`

Currently syncs an existing project based on two factors:
1. If the content of an existing file has changed, the [code](https://github.com/ansible/eda-server/blob/0cea343e310d68d8651fd55c957daf8b7c74c524/src/eda_server/project.py#L488) updates the respective DB entry. 
2. If there is a new file created, the [code](https://github.com/ansible/eda-server/blob/0cea343e310d68d8651fd55c957daf8b7c74c524/src/eda_server/project.py#L492) imports the file into its respective table (rulebook, inventory, etc.)
3. TO BE IMPLEMENTED: If the existing file's name has changed but the content is the same.

## Testing

Testing this PR mainly involves importing a Github project and syncing new changes from the project. Feel free to use the following [test project](https://github.com/Dostonbek1/test-project-for-eda).
1. Fork [this Github project](https://github.com/Dostonbek1/test-project-for-eda)
2. Create an EDA project by calling POST on `http://localhost:9000/api/projects` with the following request body:
```
{
    "url": "<link to your forked project repo>",
    "name": "Test Project",
    "description": "Test project"
}
```
3. Check the DB tables (project, rulebook, inventory, etc.) to make sure the project files have been imported.
4. Push new changes to your forked repo; could be a new rulebook, extra_var, etc. files or change the contents of the existing files.
5. Send a POST request on `http://localhost:9000/api/projects/<project_id>`.
6. Verify the above request is successful.
7. Check DB and verify all changes from the project have been synced including the `git_hash` of the project.

Resolves: AAP-5704

### CODE REVIEW

File: src/eda_server/project.py [hunk 1]
```diff
@@ -375,3 +316,238 @@ async def tar_project(db: AsyncSession, large_data_id: int, project_dir: str):
...
+        await import_rulebook(db, filename, file_content, project_id)
+
+    elif file_type == "inventory":
+        await import_inventory(db, filename, file_content, project_id)
+
+    elif file_type == "extra_var":
+        await import_extra_var(db, filename, file_content, project_id)
+
+    elif file_type == "playbook":
+        await import_playbook(db, filename, file_content, project_id)
```
Comment (cutwater) [inline src/eda_server/project.py:337]: Looks like a candidate for the "dispatch dict" pattern.

File: src/eda_server/project.py [hunk 2]
```diff
@@ -375,3 +316,238 @@ async def tar_project(db: AsyncSession, large_data_id: int, project_dir: str):
...
+            await update_inventory(db, file_content, existing_file_id)
+
+        elif file_type == "extra_var":
+            await update_extra_var(db, file_content, existing_file_id)
+
+        elif file_type == "playbook":
+            await update_playbook(db, file_content, existing_file_id)
+
+
+async def import_rulebook(
```
Comment (cutwater) [inline src/eda_server/project.py:369]: Candidates for moving into `db/sql` package

File: src/eda_server/project.py [hunk 3]
```diff
@@ -375,3 +316,238 @@ async def tar_project(db: AsyncSession, large_data_id: int, project_dir: str):
             with open(tarfile_name, "rb") as f:
                 while data := f.read(CHUNK_SIZE):
                     await lobject.write(data)
+
+
+async def import_project_file(db: AsyncSession, file, file_type, project_id):
+    directory, filename = file
+    full_path = os.path.join(directory, filename)
```
Comment (cutwater) [inline src/eda_server/project.py:323]: The fact that you pass a tuple into a function to only join it indicates that tuple is not needed to be passed to a function at all.

File: src/eda_server/project.py [hunk 4]
```diff
@@ -375,3 +316,238 @@ async def tar_project(db: AsyncSession, large_data_id: int, project_dir: str):
             with open(tarfile_name, "rb") as f:
                 while data := f.read(CHUNK_SIZE):
                     await lobject.write(data)
+
+
+async def import_project_file(db: AsyncSession, file, file_type, project_id):
```
Comment (cutwater) [inline src/eda_server/project.py:321]: Please add some type annotations. It's completely unclear what the `file` is and it's even more unexpected that this is a tuple of `(directory, filename)`.

**Review by cutwater — CHANGES_REQUESTED** [review]

**Review by ddonahue007 — APPROVED** [review]
lgtm

**Review by abaiken — APPROVED** [review]

**Pull request merged**